src/components/Button/Button.tsx

```diff
@@ -1,37 +1,74 @@
+import "./Button.scss";
 import { ReactNode } from "react";
 import cx from "classnames";
-import "./Button.scss";
+import ButtonLink from "./ButtonLink";
+
+type ButtonVariant = "primary" | "primary-action" | "semi-clear" | "clear";
 
-type Props = {
+type ButtonProps = {
   children: ReactNode;
-  imgSrc?: string;
-  imgName?: string;
+  variant: ButtonVariant;
   className?: string;
-  href?: string;
-  onClick?: () => void;
   disabled?: boolean;
+  onClick?: () => void;
+  to?: string;
+  type?: "button" | "submit" | "reset";
+  imgInfo?: {
+    src: string;
+    alt?: string;
+  };
 };
 
-export default function Button({ href, imgSrc, children, className, imgName, onClick, disabled = false }: Props) {
-  let classNames = cx("btn", className);
+export function getClassName(variant: ButtonVariant) {
+  return cx("button", {
```

Comment:
it could be replaced with just cx('button', variant)

Reply:
Also, what I usually encounter when developing UI kits, different alignment of texts and different "display" of the button. 

Usually, it is solved by props align: 'left' | 'center' and display: 'inline' | 'block' (block combines with:100% to take full width, like in SwapBox

So "variant" may only define the theme of the button, while the size and positioning are set by separate props